src/model/index.py

```diff
@@ -29,28 +29,34 @@ def serialize(self):
 
     def get_primary_keys(self, value_name):
```

Comment:
You should override this method in PrimaryIndex and make it return array with length 1 contains the primary key.
Return [value_name]